src/utils/ipfsUtils.ts

```diff
@@ -54,16 +54,52 @@ export const getCampaignData = async (ipfsKey: string): Promise<ICampaign> => {
   return data;
 };
 
-export const addStageToIpfs = (stage: IStage): IStage => {
-  return stage;
+export const addStageToIpfs = async (stage: IStage): Promise<string> => {
+  try {
+    const auth =
+      "Basic " +
+      window.btoa(
+        process.env.REACT_APP_INFURA_PROJECT_ID +
+          ":" +
+          process.env.REACT_APP_INFURA_API_SECRET
+      );
+    const client: IPFSHTTPClient = create({
+      host: "ipfs.infura.io",
+      port: 5001,
+      protocol: "https",
+      headers: {
+        authorization: auth,
+      },
+    });
+    if (client) {
+      const file = await client.add(JSON.stringify(stage));
```

Comment:
I feel that we should not be duplicating the fields that are already on the blockchain. Only store fields that are not on the blockchain to ipfs.

Reply:
Can you please specify which fields you are referring to

Reply:
goal